server/services/omnichannel-voip/service.ts

```diff
@@ -52,24 +135,120 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 		this.logger.debug({ msg: 'getAvailableExtensions()', found: filtered.length });
-		return {
-			result: filtered,
-		};
+		return filtered;
 	}
 
-	async getExtensionAllocationDetails(): Promise<IOmnichannelVoipServiceResult> {
+	async getExtensionAllocationDetails(): Promise<IAgentExtensionMap[]> {
 		const allocatedExtensions = await this.getAllocatedExtesionAllocationData({
 			username: 1,
 			roles: 1,
 			extension: 1,
 		});
 		this.logger.debug({ msg: 'getExtensionAllocationDetails() all extension length ', length: allocatedExtensions.length });
+		return allocatedExtensions.map((user: any) => ({
+			_id: user._id,
+			agentName: user.username,
+			extension: user.extension,
+		}));
+	}
+
+	/* Voip calls */
+	async findAgent(agentId: string): Promise<any> {
+		return this.normalizeAgent(agentId);
+	}
+
+	async getNewRoom(guest: ILivechatVisitor, agent: any, rid: string, roomInfo: any): Promise<IRoomCreationResponse> {
+		this.logger.debug(`Attempting to find or create a room for visitor ${guest._id}`);
+		const message = {
+			_id: Random.id(),
+			rid,
+			msg: '',
+			token: guest.token,
+			ts: new Date(),
+		};
+
+		let room = await this.voipRoom.findOneById(message.rid);
+		let newRoom = false;
+		if (room && !room.open) {
+			this.logger.debug(`Last room for visitor ${guest._id} closed. Creating new one`);
+			message.rid = Random.id();
+			room = null;
+		}
+		if (room == null) {
+			// delegate room creation to QueueManager
+			this.logger.debug(`Calling QueueManager to request a room for visitor ${guest._id}`);
+			// room = await QueueManager.requestRoom({ guest, message, roomInfo, agent: defaultAgent, extraData });
+			const name = roomInfo?.fname || guest.name || guest.username;
+			const roomId = await this.createVoipRoom(rid, name, agent, guest);
+			room = await this.voipRoom.findOneVoipRoomById(roomId);
+			newRoom = true;
+			this.logger.debug(`Room obtained for visitor ${guest._id} -> ${room?._id}`);
+		}
+		if (!room) {
+			this.logger.debug(`Visitor ${guest._id} trying to access another visitor's room`);
+			throw new Meteor.Error('cannot-access-room');
+		}
 		return {
-			result: allocatedExtensions.map((user: any) => ({
-				_id: user._id,
-				agentName: user.username,
-				extension: user.extension,
-			})),
+			room,
+			newRoom,
+		};
+	}
+
+	async findRoom(token: string, rid: string): Promise<IVoipRoom | null> {
+		const fields = {
+			t: 1,
+			departmentId: 1,
+			servedBy: 1,
+			open: 1,
+			v: 1,
+			ts: 1,
 		};
+		if (!rid) {
+			return this.voipRoom.findOneByVisitorToken(token, fields);
+		}
+		return this.voipRoom.findOneByIdAndVisitorToken(rid, token, fields);
+	}
+
+	async closeRoom(visitor: ILivechatVisitor, room: IVoipRoom, /* comment: any,*/ options = {}): Promise<boolean> {
+		this.logger.debug(`Attempting to close room ${room._id}`);
+		if (!room || room.t !== 'v' || !room.open) {
+			return false;
+		}
+
+		const params = callbacks.run('livechat.beforeCloseRoom', { room, options });
```

Comment:
I think we can comment this for now, as we're not using this callback

Reply:
Fixed. Thanks for pointing this out. There were few more errors around this.